app/models/planning_application.rb

```diff
@@ -453,6 +453,12 @@ def planning_history_enabled?
     ENV.fetch("PLANNING_HISTORY_ENABLED", "false") == "true"
   end
 
+  AssessmentDetail.categories.each_key do |category|
+    define_method(category) do
```

Comment:
Not used this before. You're dynamically creating a method for each category??

Reply:
We introduced the `assessment_details` table because a lot of new features ('Summary of works', 'Consultation summary', 'Previous applications' etc) all called for a table with `planning_application_id` and just one text field. 

So instead of planning applications having one `summary_of_work`, one `consultation_summary` etc, it has many `application_details` distinguished by category.

But it's still useful to be able to call `planning_application.summary_of_work`, instead of always calling `planning_application.application_details.find_by(category: 'summary_of_work')`.

Hope that makes sense!